naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2324,36 +2350,55 @@ impl Parser {
             types: &mut out.types,
             unresolved: &mut dependencies,
         };
+        let mut diagnostic_filters = DiagnosticFilterMap::new();
+        let ensure_no_diag_attrs =
+            |on_what_plural, filters: DiagnosticFilterMap| -> Result<(), Error> {
+                if filters.is_empty() {
+                    Ok(())
+                } else {
+                    Err(Error::DiagnosticAttributeNotSupported {
+                        on_what_plural,
+                        spans: filters.spans().collect(),
+                    })
+                }
+            };
 
         self.push_rule_span(Rule::Attribute, lexer);
         while lexer.skip(Token::Attribute) {
-            match lexer.next_ident_with_span()? {
-                ("binding", name_span) => {
+            let (name, name_span) = lexer.next_ident_with_span()?;
+            if let Some(DirectiveKind::Diagnostic) = DirectiveKind::from_ident(name) {
+                if let Some(filter) = self.diagnostic_filter(lexer)? {
+                    let span = self.peek_rule_span(lexer);
+                    diagnostic_filters.add(filter, span)?;
+                }
```

Comment:
I think this needs to break out of the loop or else it reaches `_ => return Err(Error::UnknownAttribute(name_span))`.

Reply:
I actually had discovered this with testing right after I submitted this PR, but hadn't had a chance to re-submit this. Good catch!

Fixed with the force-pushed f9ac43af.